doc/CONTRIBUTING.adoc

```diff
@@ -0,0 +1,42 @@
+= Contributing
+
+You can contribute to the modules in this repo in several ways. In terms of a "good PR" there is one important requirement to fulfill:
+
+1. A PR that provides a new class or an updated class in `parsers` module requires its respective set of unit tests.
+
+2. A PR that provides a new class or an updated class in `analyzers` module requires its respective set of unit tests.
+
+== Parsers
+
+Collector and/or ptp operator logs parsers are contributed into `src/vse_sync_pp/parsers/`.
+
+Each log file parser must be represented by a separate class that:
+
+* is provided in an appropriate module, with module named by:
+    * user space process;
+    * kernel module; or,
+    * functional area
+* is registered in `vse_sync_pp.parse.PARSERS`
+* has unit test cases built by class `ParserTestBuilder`
+    * (`tests/vse_sync_pp/parsers/test_parser.py`)
+
+== Analyzers
+
+Log file analyzers are contributed into `src/vse_sync_pp/analyzers/`.
+
+Each log file analyzer myst be represented by a separate class that:
+
+* is provided in an appropriate module
+* is registered in `vse_sync_pp.analyze.ANALYZERS`
+* has unit test cases built by class `AnalyzerTestBuilder`
+    * (`tests/vse_sync_pp/analyzers/test_analyzer.py`)
```

Comment:
Derived `TimeErrorAnalyzer` classes must override class attribute `locked`, specifying a frozenset of values representing locked states.

Also, I think we should explain what `locked` means

Reply:
Added it in the contributor guide as example.